cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -149,6 +168,24 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 	if opts.Private {
 		options = append(options, "private")
 	}
+	if opts.Overlayfs {
+		if len(opts.LowerDirs) <= 0 || len(opts.UpperDir) <= 0 || len(opts.WorkDir) <= 0 {
+			return fmt.Errorf("cannot mount %q at %q: missing arguments for overlayfs mount. lowerdir, upperdir, workdir are needed.", what, where)
+		}
+
+		var lowerDirs strings.Builder
+		for i, d := range opts.LowerDirs {
+			if i != 0 {
+				lowerDirs.WriteRune(':')
+			}
+			ed := strings.ReplaceAll(d, ",", `\,`)
```

Comment:
Please link to the kernel tree whihc handles quotiong and escaping for overlayfs.